pkg/controllers/leaderworkerset_controller.go

```diff
@@ -379,6 +412,14 @@ func (r *LeaderWorkerSetReconciler) updateConditions(ctx context.Context, lws *l
 
 				if sts.Labels[leaderworkerset.TemplateRevisionHashKey] == templateHash && leaderPod.Labels[leaderworkerset.TemplateRevisionHashKey] == templateHash {
 					updatedCount++
+
+					index, err := strconv.Atoi(sts.Labels[leaderworkerset.GroupIndexLabelKey])
+					if err != nil {
+						return false, err
+					}
+					if index < int(*lws.Spec.Replicas) {
+						currentCount++
```

Comment:
the currentCount here is the ready&updated replicas that will not be scaled down after upgrade right. If with maxSurge=2 and lws.spec.replicas=12, then once all 12 replicas is upgraded, the currentCount here is 10 and the condition will change to available based on line 437. But actually we need to perform a scale down from 12 to 10. Is there such a possibility?